ui/src/routes/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/create-run/-create-run-utils.ts

```diff
@@ -612,9 +628,74 @@ export function formValuesToPayload(
   // Reporter configuration
   //
 
+  // Check if CycloneDX, SPDX, and/or NOTICE file reports are enabled in the form,
+  // and configure them to use all output formats, accordingly.
+
+  // Unfortunately, OpenAPI's PluginConfiguration always requires 'secrets' object,
+  // and it cannot be "undefined", so we need to pass in an empty secrets object instead,
+  // which produces an eslint error about empty object types allowing non-nullish values.
+
+  /* eslint-disable @typescript-eslint/no-empty-object-type */
+
+  const cycloneDxEnabled =
+    values.jobConfigs.reporter.formats.includes('CycloneDX');
+  const spdxDocumentEnabled =
+    values.jobConfigs.reporter.formats.includes('SpdxDocument');
+  const noticeFileEnabled =
+    values.jobConfigs.reporter.formats.includes('PlainTextTemplate');
+
+  const config: {
+    SpdxDocument?: {
+      options: {
+        outputFileFormats: string;
```

Comment:
I obviously know too little about TypeSCript, but can't we replace this with something like

    outputFileFormats: spdxDocumentEnabled ? 'YAML,JSON' : string;

etc. to avoid lines 668-694?

Reply:
Ok, then I obviously really don't understand things. I though type definition was done via Zod [here](https://github.com/eclipse-apoapsis/ort-server/pull/1748/files#diff-9a8b8b3152bca6b691f8c3f876a9924e1e3f599eed2716384a259f74ce161890R37-R41).

Reply:
To clear this up, the linked Zod definition defines the type for the form visible in the UI for creating new forms. The function converts that type to format accepted by the API, so it outputs different type.